lib/veritech-server/src/handlers.rs

```diff
@@ -86,6 +87,18 @@ pub async fn process_request(
     subject: Subject,
     Headers(maybe_headers): Headers,
     msg: Message<InnerMessage>,
+) -> HandlerResult<()> {
+    metric!(counter.veritech.handlers_doing_work = 1);
+    let result = process_request_inner(state, subject, maybe_headers, msg).await;
+    metric!(counter.veritech.handlers_doing_work = -1);
+    result
```

Comment:
Now we capture the result and wait for the very possible first and last moment to increment and decrement the metric